src/insert.ts

```diff
@@ -327,7 +327,7 @@ export class InsertQuery<
               >
                 ? IsNotNull extends true
                   ? DataType | Expression<DataType, IsNotNull, any> | Query<any>
-                  : DataType | undefined | Expression<DataType, IsNotNull, any> | Query<any>
+                  : DataType | null | Expression<DataType, IsNotNull, any> | Query<any>
```

Comment:
Maybe worth defining a `type DbNull = null` somewhere and using that everywhere instead of the literal `null`?

(Right now, when I see `null`, I'm not sure if it represents a DB `null` value or if it's just one of the normal TS uses of `null`.)